views.py

```diff
@@ -106,6 +110,11 @@ def download(request, path, rest_call=False, use_async=True, *args, **kwargs):
                         response.content = "<h1>" + content_msg + "</h1>"
                     return response
 
+    elif bag_modified == "true" and metadata_dirty == 'true':
```

Comment:
This condition is for when bag_download is false, right? If so, bag_modified should NOT be checked here, but instead if metadata_dirty is true when downloading resourcemap.xml or resourcemetadata.xml, create_bag_files() needs to be called.